peps/pep-0738.rst

```diff
@@ -0,0 +1,351 @@
+apps start off as a Java virtual machine running an executable provided by the
+operating system. The app's Java code can then add native code to the process by
+loading dynamic libraries and calling them through JNI.
+
+Unlike iOS, creating subprocesses *is* supported on Android. However apps may
+only run executables in `certain locations
+<https://issuetracker.google.com/issues/128554619#comment4>`__, none of which
+are writable at runtime. Long-running subprocesses are `officially discouraged
+<https://issuetracker.google.com/issues/128554619#comment4>`__, and are not
+guaranteed to be supported in future Android versions.
+
+Android does provide a command-line shell, but this is intended only for use by
+developers, and is not available to the typical end user.
+
+For these reasons, the primary way of running Python on Android will be by
+loading ``libpython3.x.so`` into the main app process. Although there will also
+be a ``python3.x`` executable linked against ``libpython3.x.so``, this is only
+for debugging, not production use.
+
+
+Specification
+=============
+
+Scope of work
+-------------
+
+The focus of this work will be to produce an Android equivalent to the existing
+Windows embeddable package. This will be used by app developers, not end users
+directly, so no installer will be required.
+
+Adding Android as a Tier 3 platform only requires adding support for compiling
+an Android-compatible build from the unpatched CPython source code. It does not
+require there to be any officially distributed Android artefacts, although these
+could be added to python.org in the future.
+
+A Gradle project will be provided for the purpose of running the CPython test
+suite. Tooling will be provided to automate the process of building the test
+suite app, starting the emulator, installing the test suite, and executing
+it.
+
+
+Linkage
+-------
+
+For the reasons discussed in `App lifecycle`_, Python will be included in the
+app as a dynamic ``libpython3.x.so`` library. Android extension modules should
+all be linked against this library. This allows packages to use the
+``-Wl,--no-undefined`` option to detect missing symbols at build time, which can
+be a significant time-saver.
+
+An extension module linked against ``libpython3.x.so`` cannot be loaded by an
+executable that has been statically linked against ``libpython3.x.a``.
+Therefore, a static ``libpython3.x.a`` library will not be supported on Android.
+This is the same pattern used by CPython on Windows.
+
+Unlike iOS, Android allows dynamic libraries to be loaded from any location, so
+a directory tree containing co-located .py, .pyc and .so files can be handled by
+Python's standard importer.
+
+
+Standard library
+----------------
+
+A number of standard library modules will not be supported on Android because
+the underlying C APIs are not provided by the operating system:
+
+* ``crypt`` – Scheduled for removal in Python 3.13
```

Comment:
Because this PEP is targeting 3.13, We could probably remove this one as irrelevant?
```suggestion
```

Reply:
I've also removed `nis`, which is in the same situation.